entrypoint.sh

```diff
@@ -29,14 +29,49 @@ if [ ! -r "${_config_path}" ]; then
 fi
 
 # Run octodns-sync.
+_logfile="${GITHUB_WORKSPACE}/octodns-sync.log"
+_planfile="${GITHUB_WORKSPACE}/octodns-sync.plan"
+
 echo "INFO: _config_path: ${_config_path}"
 if [ "${_doit}" = "--doit" ]; then
-  script "${GITHUB_WORKSPACE}/octodns-sync.log" -e -c \
-  "octodns-sync --config-file=\"${_config_path}\" \
-  --log-stream-stdout --doit"
+  script "${_logfile}" -e -c \
+  "octodns-sync --config-file=\"${_config_path}\" --doit \
+  >>\"${_planfile}\""
 else
-  script "${GITHUB_WORKSPACE}/octodns-sync.log" -e -c \
+  script "${_logfile}" -e -c \
   "octodns-sync --config-file=\"${_config_path}\" \
-  --log-stream-stdout"
+  >>\"${_planfile}\""
 fi
 
+# If ADD_PR_COMMENT is Yes, add comment to PR.
+if [ "${ADD_PR_COMMENT}" = "Yes" ]; then
+  echo "INFO: \$ADD_PR_COMMENT is 'Yes'."
+  if [ "${GITHUB_EVENT_NAME}" = "pull_request" ]; then
+    if [ -z "${PR_COMMENT_TOKEN}" ]; then
+      echo "FAIL: \$PR_COMMENT_TOKEN is not set."
+      exit 1
+    fi
+  else
+    echo "SKIP: \$GITHUB_EVENT_NAME is not 'pull_request'."
+    exit 0
+  fi
+  # Construct the comment body
+  _sha="$(echo "${GITHUB_SHA}" | cut -c 1-7)"
+  _header="## octoDNS Plan for ${_sha}"
+  _footer="Automatically generated by octodns-sync"
+  _body="${_header}
+
+$(cat "${_planfile}")
+
+${_footer}"
+  # Post the comment
+  _user="fakename" \
+  _token="${PR_COMMENT_TOKEN}" \
+  _body="${_body}" \
+  GITHUB_EVENT_PATH="${GITHUB_EVENT_PATH}" \
+  python3 -c "import requests, os, json
+comments_url = json.load(open(os.environ['GITHUB_EVENT_PATH'], 'r'))['pull_request']['comments_url']
+response = requests.post(comments_url, auth=(os.environ['_user'], os.environ['_token']), json={'body':os.environ['_body']})
+print(response)"
```

Comment:
For posterity: This block looks the way it does, here, because it was the first way I found to meet these constraints:

- Use [the _add an issue comment_ API](https://docs.github.com/free-pro-team@latest/rest/reference/issues#create-an-issue-comment) with a provided personal access token
- Get `$_body` into the JSON payload ~safely
- Do not install `gh` or `curl`
- Do not introduce dependencies, for example an external Python script or package